runtime/queries/cpp/highlights.scm

```diff
@@ -12,56 +10,90 @@
 (template_method
   name: (field_identifier) @function)
 
-(template_function
-  name: (identifier) @function)
-
 (function_declarator
   declarator: (qualified_identifier
     name: (identifier) @function))
 
 (function_declarator
   declarator: (qualified_identifier
-    name: (identifier) @function))
+    name: (qualified_identifier
+      name: (identifier) @function)))
 
 (function_declarator
   declarator: (field_identifier) @function)
 
 ; Types
 
-((namespace_identifier) @type
- (#match? @type "^[A-Z]"))
+(namespace_definition name: (identifier) @namespace)
+(using_declaration (identifier) @namespace)
+(namespace_identifier) @namespace
 
-(auto) @type
+(qualified_identifier name: (identifier) @type.enum.variant)
 
 ; Constants
 
 (this) @variable.builtin
-(nullptr) @constant
+(nullptr) @constant.builtin
 
 ; Keywords
 
-"catch" @keyword
-"class" @keyword
-"constexpr" @keyword
-"delete" @keyword
-"explicit" @keyword
-"final" @keyword
-"friend" @keyword
-"mutable" @keyword
-"namespace" @keyword
-"noexcept" @keyword
-"new" @keyword
-"override" @keyword
-"private" @keyword
-"protected" @keyword
-"public" @keyword
-"template" @keyword
-"throw" @keyword
-"try" @keyword
-"typename" @keyword
-"using" @keyword
-"virtual" @keyword
+[
+  "catch"
+  "co_await"
+  "co_return"
+  "co_yield"
+  "concept"
+  "consteval"
+  "constinit"
+  "delete"
+  "final"
+  "noexcept"
+  "new"
+  "requires"
+  "throw"
+  "try"
+  "typename"
+  "using"
+] @keyword
+
+"<=>" @operator
+
+[
+  "and"
+  "and_eq"
+  "bitor"
+  "bitand"
+  "not"
+  "not_eq"
+  "or"
+  "or_eq"
+  "xor"
+  "xor_eq"
+] @keyword.operator
+
+[
+  "class"
+  "namespace"
+  (auto)
```

Comment:
`auto` here seems surprising, as it represents a type rather than a type category. (I would have expected it as @type).

In any case, where `auto` goes, `"decltype"` should probably go too.

Reply:
I didn't think `auto` should be `@type` directly because to me it represents something that will figure out which type it represents, but isn't name of a type itself, like an anonymous class/struct keyword. Seeing `auto` coloured as a type occasionally makes me forget there isn't (and shouldn't be) a specific struct/class named `auto`.

Reply:
TL;DR: I think `@type` *is* technically correct (and it fits my mental model better) but this isn't a big deal, go with whatever you think is most useful. Unneccesary but maybe interesting details below.

---

Microsoft is wrong I'm afraid (or rather, glossing over details). [The spec says](https://eel.is/c++draft/dcl.spec.auto.general#1)

> A [placeholder-type-specifier](https://eel.is/c++draft/dcl.spec.auto.general#nt:placeholder-type-specifier) [e.g auto] designates a [placeholder type](https://eel.is/c++draft/dcl.spec.auto.general#def:type,placeholder) that will be replaced later by deduction from an initializer[.](https://eel.is/c++draft/dcl.spec.auto.general#1.sentence-1)

The reason that `auto` can't be used in casts etc [is rather](https://eel.is/c++draft/dcl.spec.auto.general#6)

> A program that uses a placeholder type in a context not explicitly allowed in [[dcl.spec.auto]](https://eel.is/c++draft/dcl.spec.auto) is ill-formed[.](https://eel.is/c++draft/dcl.spec.auto.general#6.sentence-1)

Reply:
That fair, I've changed it back to `@type` for now unless a better reason to change it comes up